examples/cpp/hello_world/HelloWorldPublisher.h

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file HelloWorldPublisher.h
+ *
+ */
+
+#ifndef HELLO_WORLD_PUBLISHER_H_
+#define HELLO_WORLD_PUBLISHER_H_
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+
+#include "HelloWorldPubSubTypes.h"
+
+using namespace eprosima::fastdds::dds;
+
+class HelloWorldPublisher : public DataWriterListener
+{
+public:
+
+    HelloWorldPublisher();
+
+    ~HelloWorldPublisher() override;
+
+    //! Publisher matched method
+    void on_publication_matched(
+            DataWriter* writer,
+            const PublicationMatchedStatus& info) override;
+
+    //! Publish a sample
+    bool publish();
+
+    //! Run publisher
+    void run();
+
+    //! Return the current state of execution
+    static bool is_stopped();
+
+    //! Trigger the end of execution
+    static void stop();
+
+private:
+
+    HelloWorld hello_;
+
+    DomainParticipant* participant_;
+
+    Publisher* publisher_;
+
+    Topic* topic_;
+
+    DataWriter* writer_;
+
+    TypeSupport type_;
+
+    static std::atomic<bool> stop_;
```

Comment:
Why is this static?

Reply:
Because `stop()` method needs to be static